src/daft-local-execution/src/sinks/limit.rs

```diff
@@ -4,56 +4,96 @@ use common_error::DaftResult;
 use daft_micropartition::MicroPartition;
 use tracing::instrument;
 
-use super::streaming_sink::{StreamSinkOutput, StreamingSink};
+use super::streaming_sink::{StreamingSink, StreamingSinkOutput, StreamingSinkState};
+use crate::pipeline::PipelineResultType;
+
+struct LimitSinkState {
+    remaining: usize,
+}
+
+impl LimitSinkState {
+    fn new(remaining: usize) -> Self {
+        Self { remaining }
+    }
+
+    fn get_remaining(&self) -> usize {
```

Comment:
You can also implement this as `get_remaining_mut() -> &mut usize` that will give you a mutable reference to `remaining`. That way you don't have to do it as two different steps.